moveit_core/collision_detection_fcl/src/collision_common.cpp

```diff
@@ -531,14 +537,15 @@ bool distanceCallback(fcl::CollisionObjectd* o1, fcl::CollisionObjectd* o2, void
   {
     return false;
   }
-  double d = fcl::distance(o1, o2, fcl::DistanceRequestd(cdata->req->enable_nearest_points), fcl_result);
+  thread_local double distance;
+  distance = fcl::distance(o1, o2, fcl::DistanceRequestd(cdata->req->enable_nearest_points), fcl_result);
 
   // Check if either object is already in the map. If not add it or if present
   // check to see if the new distance is closer. If closer remove the existing
   // one and add the new distance information.
-  if (d < dist_threshold)
+  if (distance < dist_threshold)
   {
-    DistanceResultsData dist_result;
+    thread_local DistanceResultsData dist_result;
```

Comment:
Same here, you need to call `.clear()`.

Reply:
@AndyZe: you didn't added `.clear()` here yet. Maybe it's not needed, because all members are newly set anyway.
If so, a corresponding comment should be added.